docs/adr/0010-stop-mocking-fetch-in-tests.md

```diff
@@ -0,0 +1,36 @@
+# Replace mocking fetch with `msw`
+
+Date: 2021-01-14
+
+## Status
+
+2021-01-18 Proposed
+
+## Context
+
+For mocking unittests we are using at this moment the `jest-fetch-mock` package. This works but there are new develoments on the market. As the creator of the [Testing Library](https://testing-library.com/) suggests in this [article](https://kentcdodds.com/blog/stop-mocking-fetch), there is a new strategy available that would simplify testing of components that are making fetch calls.
+
+## Technical description
+The `msw` server intercepts the fetch calls and returns the configured response. This is done outside the test file and only overrides of the standard responses from the default configuration need to be added to the respective test file. [Here is the documentation](https://github.com/mswjs/msw).
+
+## Consideratons
+Here are few advantages of introducing this library:
+- All the fetch/mocking can moved to one place (internals/testing)
```

Comment:
What would this look like in tests that require different responses to the same call? Magic input that returns a specific response?

Reply:
Random example I found: 

`fetch.mockResponses([JSON.stringify({ detail: 'filled out' }), { status: 410 }]);`

This is already pretty succinct - how would this be changed with `msw`?

Reply:
See https://kentcdodds.com/blog/stop-mocking-fetch

Reply:
Will workout a poc asap so we solve this kind of scenarios

Reply:
Looking forward to it! Would like to doublecheck the advantages list and see how this list holds up (and if we cannot simply do it with the current solution as well)